### DISCUSSION TITLE
ATS Algo questions

### DISCUSSION THREAD

**Discussion by Aalv3** [MicroTrendsLtd/NinjaTrader8#82] [Q&A]
What file do I need to look at to see GUI Panel outputs? In other words, how can I tell what the values are when the “auto” or “t man” boxes are checked to validate my conditions? Also, if anyone knows how to implement a trailing entry as well as a “cancel if not filled within x bars”, please let me know. I can do it on a managed order but my method is not working on unmanaged

**Comment by MicroTrendsTom** [comment]
see here:
https://github.com/MicroTrendsLtd/NinjaTrader8/blob/main/ATSQuadroStrategyBase/NinjaTrader%208/bin/Custom/Strategies/ATSQuadroStategyBase/ATSQuadroStrategyBaseTradingUILayerBase.Properties.cs

and here the base

https://github.com/MicroTrendsLtd/NinjaTrader8/blob/main/ATSQuadroStrategyBase/NinjaTrader%208/bin/Custom/Strategies/AlgoSystemBase.cs   properties section are named logically for function

see line  #region properties  4120
eg IsTradeManagementEnabled

line 1467

**Comment by MicroTrendsTom** [comment]
see the samples use the methods etc
https://github.com/MicroTrendsLtd/NinjaTrader8/blob/main/ATSQuadroStrategyBase/NinjaTrader%208/bin/Custom/Strategies/ATSSamplePriceReversalTestHybridAlgo.cs

 public override void TradeManagement(double lastPrice)

**Reply by MicroTrendsTom** [reply]
![image](https://user-images.githubusercontent.com/24366913/209181055-a0fac393-6e60-4e74-983e-35d9563b0187.png)

**Reply by Aalv3** [reply]
Got it, thank you! Now will look into order cancel and trail entries

**Reply by Aalv3** [reply]
Thoughts on how to add data fields for contract size and targets/stops to the xaml panel for easy access/adjustment?

**Reply by MicroTrendsTom** [reply]
You can mod the GUI project in WPF and add fields etc?

**Reply by josecali** [reply]
Thank you, I am using "ATSQuadroStrategyBase NT8 Strategy" which is at https://ninjatraderecosystem because I have not been able to compile the GUI version.
I have Visual Studio 2022, do you recommend using an older version?

**Reply by josecali** [reply]
![Uploading Error compilacion ATS.JPG…]()

**Reply by MicroTrendsTom** [reply]
Just Follow the directions here: 
1) https://github.com/MicroTrendsLtd/NinjaTrader8/discussions/61
2)https://github.com/MicroTrendsLtd/NinjaTrader8/discussions/65
Any issues post a new issue topic but search for prior answers also as this is a very common problem resolved with the directions etc
Note here also if you want use something already built and inject your code signals - but no backtester as yet
https://algofuturestrader.com/create-your-own-rules-in-your-own-ninjatrader-scripts-and-simply-inject-them-into-the-algo-trade-entry-module/
